local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -86,23 +86,33 @@ func (d *Deleter) deletePV(pv *v1.PersistentVolume) error {
 		volMode = v1.PersistentVolumeBlock
 	}
 
-	if d.ProcTable.IsRunning(pv.Name) {
-		// Run in progress, nothing to do,
+	if d.IsCleaningInProgress(pv.Name) {
 		return nil
 	}
 
-	err = d.ProcTable.MarkRunning(pv.Name)
+	if volMode == v1.PersistentVolumeBlock && d.RuntimeConfig.UseJobForCleaning {
```

Comment:
@msau42 To clarify, if later we decide to support both block and file path as input of volume path,
the `volumeMode` cannot be a basis to determine the path type here any more (I think the field is first designed for the final format of a volume)
So maybe call `Mounter.PathIsDevice` to check the type?

Reply:
oh, I just noticed your comments https://github.com/kubernetes/community/pull/1914#pullrequestreview-110627427, again I get confused, so you mean volumeMode in PVs represent the origin type of a volume, and volumeMode in PVCs represent the its purpose type, the two can be different, is it?

Reply:
Yes, I think it would be nicer to be more explicit about the path type for cases like this.  Problem is current code only supports volumeMode in PVC and PV to be equal, so we would need to change that, but it was rejected in the initial design for the block feature.

Reply:
Looks like the current state of the proposal will not have explicit volumeMode checks.  To workaround that, I think we need explicit modes in the MountConfig.   So we can check for those modes here.

Reply:
OK. When this change is made, please make sure that the e2e tests in kubernetes are also updated. I believe they currently assume both block device and filesystem discovery folders are the same.